src/runtime/runtime.cs

```diff
@@ -1945,5 +1962,54 @@ internal static void SetNoSiteFlag()
                 }
             }
         }
+
+        /// <summary>
+        /// Return value: New reference.
+        /// </summary>
+        internal static IntPtr GetBuiltins()
+        {
+            return IsPython3 ? PyImport_ImportModule("builtins")
+                    : PyImport_ImportModule("__builtin__");
+        }
+    }
+
+
+    class PyReferenceCollection
+    {
+        public List<IntPtr> _objects { get; private set; }
+
+        public PyReferenceCollection()
+        {
+            _objects = new List<IntPtr>();
+        }
+
+        /// <summary>
+        /// Record obj's address to release the obj in the future,
+        /// obj must alive before calling Release.
+        /// </summary>
+        public void Add(ref IntPtr obj)
+        {
+            unsafe
+            {
+                fixed (void* p = &obj)
+                {
+                    _objects.Add((IntPtr)p);
+                }
+            }
+        }
+
+        public void Release()
+        {
+            foreach (var objRef in _objects)
+            {
+                unsafe
+                {
+                    var p = (void**)objRef;
```

Comment:
This looks overly complicated, and uses undefined behavior. There's no reason to assume CLR will never change addresses of those static fields, so trying to remember them for future access is a bad idea.

Also, it feels a bit overengineered. I understand the problem of accidentally forgetting to decref a field, and how zeroing fields could help debugging, but the implementation is too tricky and thus prone to errors on its own.

I think it would be quite more readable to use an ordinary `List<IntPtr>` for all objects. To zero fields you can make a holder struct for all of them, put an instance of it into `Runtime`, and simply call `Holder = default` at the end of `Shutdown`

Reply:
I just check the ECMA, the static fields is indeed moveable variable. The motivation is I want to make the assignments and action that set the member to null  in a single short line without a another hard code region. Thus a holder records the pointers is still unacceptable.
I updated the code, using lambda callbacks to set them to IntPtr.Zero, well, although it became two lines and needs a explicit assignment, the operations still in one code region(still better than separate them I think).
Seems no other better ways to achieve the object.
Since there's the only use case here, I decide not to put the `op` as the action's parameter.